src/argilla/client/feedback/metrics/agreement_metrics.py

```diff
@@ -0,0 +1,273 @@
+        self._allowed_metrics = METRICS_PER_QUESTION[self._question_type]
+
+    def compute(self, metric_names: Union[str, List[str]], **kwargs) -> List[AgreementMetricResult]:
+        """Computes the agreement metrics for the given question.
+
+        Args:
+            metric_names: name or list of names for the metrics to compute. i.e. `alpha`.
+            kwargs: additional arguments to pass to the metric.
+
+        Raises:
+            ValueError: If the metric name is not supported for the given question.
+
+        Returns:
+            agreement_metrics: A list of `AgreementMetricResult` objects for the dataset.
+        """
+        if isinstance(metric_names, str):
+            metric_names = [metric_names]
+
+        if any([metric not in self._allowed_metrics for metric in metric_names]):
+            raise ValueError(
+                f"Metrics allowed for question {self._question_name}: {list(self._allowed_metrics.keys())}"
+            )
+
+        metric_classes = [(metric_name, self._allowed_metrics[metric_name]) for metric_name in metric_names]
+
+        dataset = prepare_dataset_for_annotation_task(self._dataset, self._question_name)
+
+        distance_function = kwargs.get("distance_function")
+        if not distance_function:
+            distance_function = QUESTION_TO_DISTANCE[self._question_type]
+
+        metrics = []
+        for metric_name, metric_cls in metric_classes:
+            metric = metric_cls(annotated_dataset=dataset, distance_function=distance_function)
+            result = metric.compute(**kwargs)
+            metrics.append(AgreementMetricResult(metric_name=metric_name, result=result))
+
+        return metrics
+
+
+class AnnotationTaskMetricBase(ABC):
+    """Base class for Agreement metrics."""
+
+    def __init__(self, annotated_dataset=None, distance_function: Callable = None) -> None:
+        """
+        Args:
+            annotated_dataset: Annotated dataset as expected by the metric.
+            distance_function: Distance function to use for the metric.
+                Depending on the type of data we need a function to compute the distance.
+                For example for binary data we can use the binary distance function,
+                while RatingQuestion works with an interval distance as we are dealing with
+                numeric values.
+        """
+        self._dataset = annotated_dataset
+        self._distance_function = distance_function
+
+    def compute(self, **kwargs) -> float:
+        """General method to obtain the metric.
+
+        Args:
+            kwargs: Optional arguments that could be passed to the metric.
+
+        Returns:
+            metric: Metric result that will be stored in the `AgreementMetricResult`.
+        """
+        data = self._pre_process(self._dataset)
+        return self._compute(data, **kwargs)
+
+    def _pre_process(self, data, **kwargs) -> Any:
```

Comment:
this pre process never seems to be used or is this an NLTK internal thing? if so, perhaps mention in the docstring?